### PR TITLE
🌱 Add generate-test-infra-prowjobs makefile target and update docs

### PR DESCRIPTION
**Pull request by chrischdi**
**What this PR does / why we need it**:

Depends on:
- https://github.com/kubernetes-sigs/cluster-api/pull/9937

Related to:
- https://github.com/kubernetes/test-infra/pull/31654

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -324,7 +329,14 @@ generate-e2e-templates-v1.8: $(KUSTOMIZE)
 .PHONY: generate-e2e-templates-v1.7
 generate-e2e-templates-v1.7: $(KUSTOMIZE)
 	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_TEMPLATE_DIR)/v1.7/clusterclass" > "$(E2E_TEMPLATE_DIR)/v1.7/clusterclass-quick-start.yaml"
-	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_TEMPLATE_DIR)/v1.7/workload" > "$(E2E_TEMPLATE_DIR)/v1.7/cluster-template-workload.yaml"
+	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_TEMPLATE_DIR)/v1.7/workload" > "$(E2E_TEMPLATE_DIR)/v1.7/cluster-template-workload.yaml".PHONY: generate-test-infra-prowjobs
+
```
Comment (fabriziopandini) [inline Makefile:333]:
```suggestion
	"$(KUSTOMIZE)" --load-restrictor LoadRestrictionsNone build "$(E2E_TEMPLATE_DIR)/v1.7/workload" > "$(E2E_TEMPLATE_DIR)/v1.7/cluster-template-workload.yaml"

.PHONY: generate-test-infra-prowjobs
```

**Review by fabriziopandini — COMMENTED** [review]
one small nit, otherwise lgtm

**Comment by chrischdi** [comment]
/hold

Waiting for https://github.com/kubernetes-sigs/cluster-api/pull/10022 to merge to bump capi tools in makefile

**Comment by sbueringer** [comment]
Perfect, thx!

/approve

**Comment by chrischdi** [comment]
Thanks for the pointer, now its good 👍

**Comment by sbueringer** [comment]
Yeah my bad, I almost missed it. I should read hold comments before removing hold :)

/lgtm
/approve

**Pull request merged**